pkg/runtime/engines/singularity/config/data/singularity.conf

```diff
@@ -206,7 +206,7 @@ autofs bug path = {{$path}}
 always use nv = {{ if eq .AlwaysUseNv true }}yes{{ else }}no{{ end }}
 
 # ROOT DEFAULT CAPABILITIES: [full/file/no]
-# DEFAULT: no
+# DEFAULT: full
```

Comment:
Wouldn't it be safer to use:
```
# DEFAULT: {{ .RootDefaultCapabilities }}
```
here (and also in the other places, where possible)? This would prevent the need to hardcode the default at two different places.

Reply:
No because this template is also used at build/installation time, we parse the currently installed configuration file (in `/usr/local/etc/singularity/singularity.conf` or any configured path), use the values grabbed from the old configuration file and write a new configuration file based on the template, that means if `root default capabilities` was set to `file` in old configuration file by example we would get a `DEFAULT: file` instead of full, so we need to hardcode them. Not a big deal, changes are pretty low in this file

Reply:
> we parse the currently installed configuration file (in `/usr/local/etc/singularity/singularity.conf` or any configured path), use the values grabbed from the old configuration file and write a new configuration file based on the template,

Do I understand correctly that this means that when installing singularity on a system on which singularity was already installed and configured before, the generated and installed config file will look different (i.e. similar to the old one) and not be `vanilla` anymore? 

Is there a way to have singularity install itself with the vanilla / "upstream" default configuration independent of what is present on the system? 
Otherwise, this breaks configuration update systems of downstream Linux distributions, since the config file of a "second" installation looks different than the original config file of the first installation before user modifications, and users will not be alerted of changed defaults.

Reply:
Yes but this behaviour can be disabled for package maintainers

To take old values:

```
./mconfig
make -C builddir
...
```

For a vanilla configuration:
```
./mconfig
make old_config= -C builddir
...
```

Reply:
Thanks, that explains it very well :smile:.